contentcuration/contentcuration/views.py

```diff
@@ -275,55 +277,98 @@ def duplicate_nodes(request):
             "node_ids": " ".join(new_nodes)
         }))
 
-def _duplicate_node(node, sort_order=None, parent=None, channel_id=None):
+def _duplicate_node_bulk(node, sort_order=None, parent=None, channel_id=None):
+
+    if isinstance(node, int) or isinstance(node, basestring):
+        node = ContentNode.objects.get(pk=node)
+
+    # keep track of the in-memory models so that we can bulk-create them at the end (for efficiency)
+    to_create = {
+        "nodes": [],
+        "node_files": [],
+        "assessment_files": [],
+        "assessments": [],
+    }
+
+    # perform the actual recursive node cloning
+    new_node = _duplicate_node_bulk_recursive(node=node, sort_order=sort_order, parent=parent, channel_id=channel_id, to_create=to_create)
+
+    # create nodes, one level at a time, starting from the top of the tree (so that we have IDs to pass as "parent" for next level down)
+    for node_level in to_create["nodes"]:
+        for node in node_level:
+            node.parent_id = node.parent.id
+        ContentNode.objects.bulk_create(node_level)
+        for node in node_level:
+            for tag in node._meta.tags_to_add:
+                node.tags.add(tag)
+
+    # rebuild MPTT tree for this channel (since we're inside "disable_mptt_updates", and bulk_create doesn't trigger rebuild signals anyway)
+    ContentNode.objects.partial_rebuild(to_create["nodes"][0][0].tree_id)
+
+    # create each of the assessment items
+    for a in to_create["assessments"]:
+        a.contentnode_id = a.contentnode.id
+    AssessmentItem.objects.bulk_create(to_create["assessments"])
```

Comment:
It... worked! AutoField makes it easier, as the DB handles it. It was the UUID ones working that surprised me.